lib/symengine/basic.rb

```diff
@@ -11,8 +11,13 @@ def free_symbols
     end
     
     def abs
-        return SymEngine::abs(self)
+      SymEngine::abs(self)
+    end
+    def to_proc()
+      if self.free_symbols.count > 1
+        raise ArgumentError, "Too many free symbols! Only 1 allowed. Found #{self.free_symbols.count}."
```

Comment:
SymEngine doesn't track the order in which the formula is created. `PI*r**2*h` and `PI*h*r**2`  both have the same representation in SymEngine and are indistinguishable.
cc @certik